packages/site/src/layout/NavMenu.module.css

```diff
@@ -0,0 +1,96 @@
+.navMenuContainer {
+    --hoverOffset: var(--space-smallest);
+    --negativeHoverOffset: calc(-1 * var(--hoverOffset));
+    --subItemInset: calc(var(--space-small) + var(--space-smaller));
+    --navItemHeight: 40px;
+    display: flex;
+    flex-direction: column;
+    padding: 36px 0 0 0;
+    height: 100dvh;
+    background-color: var(--color-surface--background);
+    width: 200px;
+    box-sizing: border-box;
+}
+
+.navMenuContainer > * {
+  padding-left: var(--space-base);
+  padding-right: var(--space-base);
+  padding-bottom: var(--space-base);
+}
+
+.navMenuHeader {
+    display: flex;
+    flex-direction: column;
+    gap: var(--space-large);
+}
+
+.navMenu {
+    color: var(--color-text);
+    display: flex;
+    flex-direction: column;
+    gap: var(--space-large);
+    overflow-y: scroll;
+}
+
+.navMenu li {
+  margin-bottom: var(--space-smaller);
+}
+
+.navMenuItem {
+  align-items: center;
+  display: flex;
+  min-height: var(--navItemHeight);
+  overflow-wrap: anywhere;
+  transition: all var(--timing-base) ease-out;
+  user-select: none;
+}
+
+.navMenuLink {
+  color: var(--color-text);
+  outline: transparent;
+  text-decoration: none;
+  padding: 0 var(--space-smallest);
+
+  &:visited {
+    color: var(--color-text);
+  }
+
+  &:hover,
+  &:focus-visible {
+    background-color: var(--color-surface--background--hover);
+    color: var(--color-text);
+  }
+}
+
+.navMenuSubItem {
+  padding-left: var(--subItemInset);
+}
+
+.navMenuSubTitle {
+  min-height: 24px;
+  padding: var(--space-smaller) 0 var(--space-small) 0;
+  align-items: end;
+}
+
+.disclosureNavItem {
+  display: flex;
+  gap: var(--space-smaller);
+  border-radius: var(--radius-small);
+  transition: all var(--timing-base) ease-out;
+  margin-left: var(--negativeHoverOffset);
+  margin-right: var(--negativeHoverOffset);
+  padding-left: var(--hoverOffset);
+  padding-right: calc(var(--space-smallest) + var(--hoverOffset));
```

Comment:
Bit of a tricky means to have the rest state align with the overall structure, while avoiding the hover state touching the label / button boundaries